pkg/crc/machine/kubeconfig.go

```diff
@@ -264,19 +271,57 @@ func mergeConfigHelper(kubeConfigFile, globalConfigFile string) error {
 	if err != nil {
 		return err
 	}
+	// append cluster name to the AuthInfos
+	cfg, err := appendClusterToAuthinfos(currentConf)
+	if err != nil {
+		return err
+	}
 	// Merge the currentConf to globalConfig
-	for name, cluster := range currentConf.Clusters {
+	for name, cluster := range cfg.Clusters {
 		globalConf.Clusters[name] = cluster
 	}
 
-	for name, authInfo := range currentConf.AuthInfos {
+	for name, authInfo := range cfg.AuthInfos {
 		globalConf.AuthInfos[name] = authInfo
 	}
 
-	for name, context := range currentConf.Contexts {
+	for name, context := range cfg.Contexts {
 		globalConf.Contexts[name] = context
 	}
 
-	globalConf.CurrentContext = currentConf.CurrentContext
+	globalConf.CurrentContext = cfg.CurrentContext
 	return clientcmd.WriteToFile(*globalConf, globalConfigPath)
 }
+
+func appendClusterToAuthinfos(cfg *api.Config) (*api.Config, error) {
+	var clusterURL string
+	for _, cluster := range cfg.Clusters {
+		clusterURL = cluster.Server
```

Comment:
shouldn't this `clusterAPI` because below in the function you used `clusterURL` for string replace?

Reply:
Apart from this lgtm.

Reply:
`clusterURL` here and  `clusterAPI` has the same value, do you mean just to name the variable as `clusterAPI`, or actually use the `ClusterConfig.ClusterAPI`?

Reply:
i wanted to avoid having to pass the ClusterConfig to this function so deriving the URL from the `clusters` section of the kubeconfig itself

Reply:
@anjannath I was only referring the variable name not `ClusterConfig.ClusterAPI`